### PR TITLE
MODEXPW-386 - "Connection reset (SocketException)" error bulk editing Items on large bulk edit job

### PR DESCRIPTION
**Pull request by siarhei-charniak**
[MODEXPW-386](https://issues.folio.org/browse/MODEXPW-386) - "Connection reset (SocketException)" error bulk editing Items on large bulk edit job

## Approach
Improved logic

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [ ] Check logging
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/dew/batch/JobCompletionNotificationListener.java [hunk 1]
```diff
@@ -174,6 +178,14 @@ private void processJobAfter(String jobId, JobParameters jobParameters) {
       acknowledgementRepository.deleteAcknowledgement(jobId);
     }
 
+    if (BULK_EDIT_IDENTIFIERS.getValue().equals(jobParameters.getString(EXPORT_TYPE)) && nonNull(jobParameters.getString(FILE_NAME))) {
+      try {
+        Files.deleteIfExists(Path.of(jobParameters.getString(TEMP_IDENTIFIERS_FILE_NAME)));
+      } catch (IOException e) {
+        // do nothing
```
Comment (TarasSpashchenko) [inline src/main/java/org/folio/dew/batch/JobCompletionNotificationListener.java:185]: Please add a log warning here

**Review by TarasSpashchenko — CHANGES_REQUESTED** [review]

File: src/main/java/org/folio/dew/batch/JobCompletionNotificationListener.java (hunk 1 above)
Comment (siarhei-charniak) [reply src/main/java/org/folio/dew/batch/JobCompletionNotificationListener.java:174]: Good catch, thanks!

**Review by TarasSpashchenko — APPROVED** [review]

**Comment by siarhei-charniak** [comment]
Screenshot 2023-03-21 at 15 42 57

File: src/test/java/org/folio/dew/BulkEditTest.java [hunk 2]
```diff
@@ -727,6 +729,11 @@ private JobParameters prepareJobParameters(ExportType exportType, EntityType ent
       parametersBuilder.addLong(TOTAL_CSV_LINES, countLines(localFilesStorage, file, false), false);
     }
 
+    var tempDir = System.getProperty(TMP_DIR_PROPERTY) + PATH_SEPARATOR + springApplicationName + PATH_SEPARATOR + jobId;
```
Comment (alekGbuz) [inline src/test/java/org/folio/dew/BulkEditTest.java:732]: Plz, use Path options to create path for directory, path separator in that case will be operational system independent.

File: src/main/java/org/folio/dew/controller/BulkEditController.java [hunk 3]
```diff
@@ -334,6 +342,16 @@ public ResponseEntity<String> uploadCsvFile(UUID jobId, MultipartFile file) {
     }
   }
 
+  private String saveTemporaryIdentifiersFile(UUID jobId, MultipartFile file) throws IOException {
+    var tempDir = System.getProperty(TMP_DIR_PROPERTY) + PATH_SEPARATOR + springApplicationName + PATH_SEPARATOR + jobId;
```
Comment (alekGbuz) [inline src/main/java/org/folio/dew/controller/BulkEditController.java:346]: Plz, use Path options to create path for directory, path separator in that case will be operational system independent.

**Review by obozhko-folio — APPROVED** [review]

File: src/test/java/org/folio/dew/controller/BulkEditControllerTest.java [hunk 4]
```diff
@@ -1333,6 +1333,7 @@ void shouldReplaceEmailAddress() {
   @Test
   @DisplayName("Post user content update to replace patron group and replace expiration date")
   @SneakyThrows
+  @Disabled
```
Comment (khandramai) [inline src/test/java/org/folio/dew/controller/BulkEditControllerTest.java:1336]: Why this test is disabled?

**Pull request merged**